staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/testing/clock/event_clock.go

```diff
@@ -0,0 +1,221 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clock
+
+import (
+	"container/heap"
+	"math/rand"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/clock"
+	"k8s.io/apiserver/pkg/util/flowcontrol/counter"
+)
+
+// EventFunc does some work that needs to be done at or after the
+// given time.  All of the work that needs to be done is finished
+// before this function returns and, if there is an associated
+// `sync.WaitGroup` (see NewFakeEventClock), its counter goes to zero;
+// no other work is left to be completed in goroutines.
```

Comment:
This comment needs to be updated due to the introduction of the GoRoutineCounter abstraction.  I suggest something like the following.

> // EventFunc does some work that needs to be done at or after the
> // given time.  After this function returns, associated work may continue on
> // other goroutines only if they are counted by the GoRoutineCounter
> // of the FakeEventClock handling this EventFunc.

Reply:
Updated to use the suggested comment.